src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoWhenNewToMockito.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesMethod<>(PM_WHEN_NEW), new JavaVisitor<ExecutionContext>() {
+            @Override
+            public @Nullable J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                if (THEN_RETURN.matches(method) && method.getSelect() instanceof J.MethodInvocation) {
+                    J.MethodInvocation select1 = (J.MethodInvocation) method.getSelect();
+                    if (WITH_NO_ARGUMENTS.matches(select1) && select1.getSelect() instanceof J.MethodInvocation) {
+                        J.MethodInvocation select2 = (J.MethodInvocation) select1.getSelect();
+                        if (PM_WHEN_NEW.matches(select2) && select2.getArguments().size() == 1) {
+                            maybeRemoveImport("org.powermock.api.mockito.PowerMockito");
+
+                            Cursor containingMethod = getCursor().dropParentUntil(x -> x instanceof J.MethodDeclaration);
+                            Expression argument = select2.getArguments().get(0);
+                            if (argument instanceof J.FieldAccess) {
+                                ArrayList<J.FieldAccess> listOfMocks = containingMethod.getMessage("POWERMOCKITO_WHEN_NEW_REPLACED", new ArrayList<J.FieldAccess>());
+                                listOfMocks.add((J.FieldAccess) argument);
+                                containingMethod.putMessage("POWERMOCKITO_WHEN_NEW_REPLACED", listOfMocks);
+                                return null;
+                            }
+                        }
+                    }
+                }
+                return super.visitMethodInvocation(method, ctx);
+            }
+
+            @Override
+            public J visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                J ret = super.visitMethodDeclaration(method, ctx);
+                List<J.FieldAccess> mockArguments = getCursor().getMessage("POWERMOCKITO_WHEN_NEW_REPLACED");
+                if (mockArguments != null && ret instanceof J.MethodDeclaration) {
+                    doAfterVisit(removeMockUsagesVisitor(mockArguments, method.toString()));
+
+                    J.MethodDeclaration retM = (J.MethodDeclaration) ret;
+
+                    // onlyIfReferenced=false as `maybeAddImport` doesn't seem to find the type referred to in a try statement
+                    // see https://github.com/openrewrite/rewrite/issues/5187
+                    maybeAddImport("org.mockito.MockedConstruction", false);
+
+                    for (J.FieldAccess mockArgument: mockArguments) {
+                        String mockedClassName = ((J.Identifier) mockArgument.getTarget()).getSimpleName();
+                        String variableNameForMock = VariableNameUtils.generateVariableName("mock" + mockedClassName, updateCursor(ret), VariableNameUtils.GenerationStrategy.INCREMENT_NUMBER);
```

Comment:
Nitpick: I would use a static import for readability:

```suggestion
                        String variableNameForMock = VariableNameUtils.generateVariableName("mock" + mockedClassName, updateCursor(ret), INCREMENT_NUMBER);
```

or even:

```suggestion
                        String variableNameForMock = generateVariableName("mock" + mockedClassName, updateCursor(ret), INCREMENT_NUMBER);
```

Reply:
7e690a8